### PR TITLE
Enhance ABAP Language Version class

### PR DESCRIPTION
**Pull request by mbtools**
- Class is now based on repo settings instead of repo instance
- Toggled by experimental feature and repo setting (see below)
- Add method to get repo setting (to be used for serializing objects)
- Always allow importing new (non-existing) packages
- Differentiate between source code object and non-source code objects
- Lots of unit tests

Ref #6154, #6298

### CODE REVIEW

**Comment by mbtools** [comment]
@larshp Tests for `zcl_abapgit_abap_language_vers` won't work in abaplint since they depend on SAP classes. However, I don't see the class included in [abap_transpile](https://github.com/abapGit/abapGit/blob/main/test/abap_transpile.json). Why is it trying to run these tests?

File: src/utils/zcl_abapgit_abap_language_vers.clas.abap [hunk 1]
```diff
@@ -5,34 +5,51 @@ CLASS zcl_abapgit_abap_language_vers DEFINITION
 
   PUBLIC SECTION.
 
+    CONSTANTS c_any_abap_language_version TYPE zif_abapgit_aff_types_v1=>ty_abap_language_version VALUE '*'.
     CONSTANTS c_feature_flag TYPE string VALUE 'ALAV'.
 
+    METHODS constructor
+      IMPORTING
+        !io_dot_abapgit TYPE REF TO zcl_abapgit_dot_abapgit.
```
Comment (larshp) [inline src/utils/zcl_abapgit_abap_language_vers.clas.abap:13]: can this be changed to `zif_abapgit_dot_abapgit` instead?

**Review by larshp — CHANGES_REQUESTED** [review]
see below

File: src/utils/zcl_abapgit_abap_language_vers.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/utils/zcl_abapgit_abap_language_vers.clas.abap:13]: `zif_abapgit_dot_abapgit` contains only constants. so that would be a bigger change

Comment (larshp) [reply src/utils/zcl_abapgit_abap_language_vers.clas.abap:13]: yea, okay

**Review by larshp — APPROVED** [review]

**Pull request merged**